airbyte-commons/src/main/java/io/airbyte/commons/util/MoreIterators.java

```diff
@@ -0,0 +1,236 @@
+    while (iterator.hasNext()) {
+      list.add(iterator.next());
+    }
+    return list;
+  }
+
+  /**
+   * Create a set from an iterator
+   *
+   * @param iterator iterator to convert
+   * @param <T> type
+   * @return set
+   */
+  public static <T> Set<T> toSet(Iterator<T> iterator) {
+    final Set<T> set = new HashSet<>();
+    while (iterator.hasNext()) {
+      set.add(iterator.next());
+    }
+    return set;
+  }
+
+  /**
+   * Create an iterator that only executes the given supplier after the first invocation of hasNext.
+   * When fully consumed it will close the input stream.
+   *
+   * @param streamSupplier stream to be supplied lazily
+   * @param <T> type
+   * @return lazy, auto closing iterator
+   */
+  public static <T> Iterator<T> streamSupplierToLazyIterator(Supplier<Stream<T>> streamSupplier) {
+    return iteratorSupplierToLazyIterator(() -> autoCloseIterator(streamSupplier.get()));
+  }
+
+  /**
+   * Create an iterator that only executes the given supplier after the first invocation of hasNext.
+   *
+   * @param iteratorSupplier iterator to be supplied lazily
+   * @param <T> type
+   * @return lazy iterator
+   */
+  public static <T> Iterator<T> iteratorSupplierToLazyIterator(Supplier<Iterator<T>> iteratorSupplier) {
+    return new LazyIterator<>(iteratorSupplier);
+  }
+
+  /**
+   * Creates an iterator from an iterator. The new iterator when fully consumed will execute the
+   * onClose function.
+   *
+   * @param iterator input iterator
+   * @param onClose on close function to be executed when iterator is completely consumed
+   * @param <T> type
+   * @return auto closing iterator
+   */
+  public static <T> Iterator<T> autoCloseIterator(Iterator<T> iterator, SafeVoidCallable onClose) {
+    return new AutoCloseIterator<>(iterator, onClose);
+  }
+
+  /**
+   * Takes a stream and turns it into an iterator that when fully consumed will close the stream.
+   *
+   * @param stream stream to convert
+   * @param <T> type
+   * @return auto closing iterator
+   */
+  public static <T> Iterator<T> autoCloseIterator(Stream<T> stream) {
+    return new AutoCloseIterator<>(stream.iterator(), stream::close);
+  }
+
+  /**
+   * Coerces a vanilla {@link Iterator} into a {@link CloseableIterator} by adding a no op close
+   * function.
+   *
+   * @param iterator iterator to convert
+   * @param <T> type
+   * @return closeable iterator
+   */
+  public static <T> CloseableIterator<T> toCloseableIterator(Iterator<T> iterator) {
+    return new DefaultCloseableIterator<>(iterator, new VoidCallable.NoOp());
+  }
+
+  /**
+   * Coerces a {@link CloseableIterator} into a vanilla {@link Iterator}. When fully consumed, the
+   * output iterator will call the close function of the input {@link CloseableIterator}.
+   *
+   * @param iterator closeable iterator
+   * @param <T> type
+   * @return vanilla iterator
+   */
+  public static <T> Iterator<T> toCloseableIterator(CloseableIterator<T> iterator) {
```

Comment:
you need to kill this function, it cannot exist. Way too dangerous for the reason I mentioned earlier

Reply:
removed.